src/client/tile.cpp

```diff
@@ -386,14 +386,21 @@ class TextureSource : public IWritableTextureSource
 
 	// Generate image based on a string like "stone.png" or "[crack:1:0".
 	// if baseimg is NULL, it is created. Otherwise stuff is made on it.
-	bool generateImagePart(std::string part_of_name, video::IImage *& baseimg);
+	bool generateImagePart(std::string part_of_name, video::IImage *& baseimg,
+		bool normalmap_found);
 
+	enum normal_presence {
+		NORMAL_ABSENT,
+		NORMAL_PRESENT,
+		NORMAL_UNKNOWN,
+	};
 	/*! Generates an image from a full string like
 	 * "stone.png^mineral_coal.png^[crack:1:0".
 	 * Shall be called from the main thread.
 	 * The returned Image should be dropped.
 	 */
-	video::IImage* generateImage(const std::string &name);
+	video::IImage* generateImage(const std::string &name,
+		normal_presence is_normalmap = NORMAL_UNKNOWN);
```

Comment:
Pointer misplaced (touched line)
`is_normalmap` sounds like a boolean value, but it's an enum. How about `image_type` or `image_info`?

Reply:
I've called it normalmap_info and fixed the * position.